crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -829,17 +829,20 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
         // canonical, but in the db. If it is in a sidechain, it is not canonical. If it is not in
         // the db, then it is not canonical.
 
+        let factory = self.externals.database();
```

Comment:
I'm not a fan of the term factory here, because this doesn't represent anything we have here.
imo we should just call this var `db`

Reply:
nvm with #3121 this rename makes sense